kev/backends/dynamodb/db.py

```diff
@@ -0,0 +1,214 @@
+import boto3
+import json
+import re
+import logging
+from kev.backends import DocDB
+
+from pprint import pprint
+from boto3.dynamodb.conditions import Key
+
+
+class DynamoDB(DocDB):
+
+    db_class   = boto3.resource
+    backend_id = 'dynamodb'
+    doc_id_string = '{doc_id}:id:{backend_id}:{class_name}'
+    index_pattern = r'^(?P<backend_id>[-\w]+):(?P<class_name>[-\w]+)' \
+                    ':indexes:(?P<index_name>[-\w]+):(?P<index_value>' \
+                    '[-\W\w\s]+)/(?P<doc_id>[-\w]+):id:' \
+                    '(?P<backend_id_b>[-\w]+):(?P<class_name_b>[-\w]+)$'
+
+
+    ############################################################################
+    # Desc: dynamodb DB constructor
+    #
+    # @param  kwargs[aws_secret_access_key]; aws secret key
+    # @param  kwargs[aws_access_key_id]; aws acess key id
+    # @param  kwargs[table]; aws dynamodb table name
+    ############################################################################
+    def __init__(self,**kwargs):
+        # FUTURE: create helper methods to create/delete tables(useful for testing)
+
+        if 'aws_secret_access_key' in kwargs and 'aws_access_key_id' in kwargs:
+            boto3.Session(aws_secret_access_key=kwargs['aws_secret_access_key'],
+                aws_access_key_id=kwargs['aws_access_key_id'])
+
+        self._db      = boto3.resource('dynamodb')
+        self._client  = boto3.client('dynamodb')
+        self.table    = kwargs['table']
+        self._indexer = self._db.Table(self.table)
+
+        # setup logger
+        self.logName  = str(self.__class__.__name__)
```

Comment:
In Python we don't camel case variables​, functions, methods, or classes. We use underscores for variables, functions, and methods. We format classes like this MyClassName.